### PR TITLE
Og terra notification

### PR DESCRIPTION
**Pull request by GarimaDamani**

### CODE REVIEW

File: opsgenie/provider.go [hunk 1]
```diff
@@ -43,6 +44,7 @@ func Provider() terraform.ResourceProvider {
 			"opsgenie_escalation": dataSourceOpsgenieEscalation(),
 			"opsgenie_schedule":   dataSourceOpsgenieSchedule(),
 			"opsgenie_heartbeat":  dataSourceOpsgenieHeartbeat(),
+			//"opsgenie_notification_rule": dataSourceOpsgenieNotificationRule(),
```
Comment (ffahri) [inline opsgenie/provider.go]: we can remove that

File: opsgenie/resource_opsgenie_notification_rule_test.go [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      method = "email"
+      to = "genieuser-%s@opsgenie.com"
+    }
+  }
+  order = 0
+  enabled = true
+  repeat {
+    loop_after = 2
+  }
+  timeRestriction {
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_notification_rule_test.go]: should be time_restriction

**Review by ffahri — CHANGES_REQUESTED** [review]
e2e test are failing

File: opsgenie/provider.go (hunk 1 above)
Comment (GarimaDamani) [reply opsgenie/provider.go]: yes

File: opsgenie/resource_opsgenie_notification_rule_test.go (hunk 2 above)
Comment (GarimaDamani) [reply opsgenie/resource_opsgenie_notification_rule_test.go]: Let me fix it. and run the test.

File: opsgenie/resource_opsgenie_notification_rule_test.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+    restriction {
+      start_hour = 3
+      start_min = 15
+      end_hour = 5
+      end_min = 30
+    }
+  }
+}
+
+resource "time_sleep" "wait_40_seconds" {
```
Comment (ffahri) [inline opsgenie/resource_opsgenie_notification_rule_test.go]: why do we need this resource, created resource is not available asap ?

File: opsgenie/provider.go (hunk 1 above)
Comment (GarimaDamani) [reply opsgenie/provider.go]: Done. removed. resolving the conversation.

File: opsgenie/resource_opsgenie_notification_rule_test.go (hunk 3 above)
Comment (GarimaDamani) [reply opsgenie/resource_opsgenie_notification_rule_test.go]: We were trying to check how destroy function works. Removed. Resolving the conversation.

File: opsgenie/resource_opsgenie_notification_rule_test.go (hunk 2 above)
Comment (GarimaDamani) [reply opsgenie/resource_opsgenie_notification_rule_test.go]: Fixed. tests are passing. resolving the conversation.

**Comment by GarimaDamani** [comment]
> e2e test are failing

Looks like they are good now.

**Review by ffahri — APPROVED** [review]

**Pull request merged**